index.js

```diff
@@ -47,8 +52,13 @@ HandlebarsCompiler.prototype.compile = function(data, path, callback) {
   var error, key, ns, result, source;
   try {
     source = "Handlebars.template(" + (handlebars.precompile(data)) + ")";
-    if (this.namespace) {
+
+    if(this.isFunction(this.namespace))
```

Comment:
`typeof this.namespace === 'function'` should be sufficient

and please match code style: `if (cond)` with whitespace, and use brackets for multi-line `if`s